Update documentation to clarify --watch behavior.

astegmaier opened the pull request:
## Summary

In a recent project, I was totally stumped about how to get `jest --watch` working. The first run was fine, but it wouldn't re-run automatically in response to test file changes. (I verified that `git` could see the changes fine, and there were no visible errors). After two days of debugging/searching, I discovered that the reason: due to [this line](https://github.com/facebook/jest/blob/9ac2dcd55c0204960285498c590c1aa7860e6aa8/packages/jest-haste-map/src/index.ts#L826), the watcher will always ignore files and folders starting with a dot (`.`). (The project I inherited for some reason had all the tests in a folder called `.tests`).

As far as I can tell this isn't documented anywhere on the internet, and I though it would be a nice thing to do to put something out there to clarify and save other people the headache.

The contributing guide says that "significant changes to the documentation or website" should be documented in CHANGLOG.md, but I wouldn't call this 'significant' (although I'm happy to add an entry if anyone disagrees).

## Test plan

This just adds a sentence to the documentation, and I didn't think testing was necessary beyond spell checking and making sure it looked good in my markdown editor.

SimenB commented on docs/Configuration.md:
```diff
@@ -1135,6 +1135,8 @@ An array of RegExp patterns that are matched against all source file paths befor
 
 These patterns match against the full path. Use the `<rootDir>` string token to include the path to your project's root directory to prevent it from accidentally ignoring all of your files in different environments that may have different root directories. Example: `["<rootDir>/node_modules/"]`.
 
+Even if nothing is specified here, the watcher will ignore changes to any file and folder that begins with a dot (`.`).
```
```suggestion
Even if nothing is specified here, the watcher will ignore changes to any hidden files and directories, i.e. files and folders that begins with a dot (`.`).
```

or something to that effect.

astegmaier replied:
Done!

astegmaier commented:
I've updated the PR in response to the feedback (i.e. also changed the versioned docs and tweaking the message text). Let me know if I can do anything else on my end.

SimenB commented:
@astegmaier sorry, this got away from me over the holidays 😬 Landed in f47ade4ed
